configs/.eslintrc.yml

```diff
@@ -10,18 +10,69 @@ plugins: [
     '@typescript-eslint'
 ]
 
-"rules":
-   "import/extensions": [
-      "error",
-      "ignorePackages",
-      {
+rules:
+    "import/extensions": [
+        "error",
+        "ignorePackages",
+        {
         "js": "never",
         "jsx": "never",
         "ts": "never",
         "tsx": "never"
-      }
-   ]
-
+        }
+    ]
+    "no-unused-vars": "off"
+    "@typescript-eslint/no-unused-vars": ["error", { "ignoreRestSiblings": true }]
+    # Temp turn off following rules because we're doing ts-migrate
+    # Will turn them on after migration completed.
+    "max-len": "warn"
+    "no-use-before-define": "warn"
```

Comment:
這就是在 `ts-migrate` 後暫時加入的 rule，理由是 ts-migrate 會加入太長的 `@ts-expect-error` comment，並且有產生一些會提前使用 type 的 code(違反 `no-use-before-define`)。

Reply:
稍微補充說明一下 `@ts-expect-error` 的用途：

The 3.9 release of TypeScript introduced @ts-expect-error comments. When a line is prefixed with a @ts-expect-error comment, TypeScript will suppress that error from being reported. If there’s no error, TypeScript will report that @ts-expect-error wasn’t necessary. In the Airbnb codebase we switched to using the @ts-expect-error comment instead of @ts-ignore.

ref: https://medium.com/airbnb-engineering/ts-migrate-a-tool-for-migrating-to-typescript-at-scale-cd23bfeb5cc